testbed/testbed/validator.go

```diff
@@ -36,12 +36,43 @@ type TestCaseValidator interface {
 }
 
 // PerfTestValidator implements TestCaseValidator for test suites using PerformanceResults for summarizing results.
-type PerfTestValidator struct{}
+type PerfTestValidator struct {
+
+	// IsScraping specifies if the load generator
+	IsScraping bool
+
+	// ScrapeInterval is the interval the receiver is scraping at, if IsScraping is enabled
+	ScrapeInterval time.Duration
+
+	// ExtraItemsScrapedPerIntervalCount accounts for if the receiver adds additional metrics than those sent
+	// such as the Prometheus 'up' metric
+	ExtraItemsScrapedPerIntervalCount int
+}
 
 func (v *PerfTestValidator) Validate(tc *TestCase) {
+	expectedDataItemsReceived := tc.MockBackend.DataItemsReceived()
+
+	if v.IsScraping {
+
+		// Get expected amount if additional metrics are added after scrape
+		expectedTotalExtraItemsReceived := (uint64(len(tc.MockBackend.ReceivedTimestamps) * v.ExtraItemsScrapedPerIntervalCount))
+		expectedDataItemsReceived = tc.MockBackend.DataItemsReceived() - expectedTotalExtraItemsReceived
```

Comment:
Unclear what this calculation does, why the multiplication, then subtraction. Explain the logic in a comment.
I may be wrong but I think the names probably are not reflecting the intent, that may be contributing to the confusion. Perhaps use more descriptive names for variables.